usecase/src/test/java/org/mobilitydata/gtfsvalidator/usecase/ValidateStopTimeIncreasingDistanceTest.java

```diff
@@ -0,0 +1,321 @@
+/*
+ *  Copyright (c) 2020. MobilityData IO.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.mobilitydata.gtfsvalidator.usecase;
+
+import org.junit.jupiter.api.Test;
+import org.mobilitydata.gtfsvalidator.domain.entity.gtfs.stoptimes.StopTime;
+import org.mobilitydata.gtfsvalidator.domain.entity.notice.error.DecreasingStopTimeDistanceErrorNotice;
+import org.mobilitydata.gtfsvalidator.domain.entity.notice.warning.DecreasingStopTimeDistanceWarningNotice;
+import org.mobilitydata.gtfsvalidator.usecase.port.ValidationResultRepository;
+import org.mockito.ArgumentCaptor;
+
+import java.util.TreeMap;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mobilitydata.gtfsvalidator.domain.entity.notice.base.Notice.*;
+import static org.mockito.Mockito.*;
+
+class ValidateStopTimeIncreasingDistanceTest {
+
+    // suppressed warning regarding ignored result of method used in assertions
+    @SuppressWarnings("ResultOfMethodCallIgnored")
```

Comment:
I would add one more test where the first stop sequence shape_dist_traveled is 10 and the second 5, just to cover all corner cases (i.e., make sure a large value in the first sequence index is detected).